aspnetcore/signalr/get-started-signalr-core.md

```diff
@@ -78,24 +120,66 @@ The SignalR server must be configured so that it knows to pass requests to Signa
 
   [!code-csharp[Startup](get-started-signalr-core/sample/Startup.cs?highlight=22,40-43)]
 
+# [Visual Studio Code](#tab/visual-studio-code)
+
+1. To configure a SignalR project, modify the `ConfigureServices` method of the application's `Startup` class by inserting a call to `services.AddSignalR`.
+
+  `services.AddSignalR` adds SignalR as part of the [ASP.NET Core middleware](xref:fundamentals/middleware/index) pipeline.
+
+1. Configure routes to your hubs using `UseSignalR`.
+
+  [!code-csharp[Startup](get-started-signalr-core/sample/Startup.cs?highlight=22,40-43)]
+
+-----
+
 ## Create the SignalR client code
 
+# [Visual Studio](#tab/visual-studio)
+
 1. Replace the content in *Pages\Index.cshtml* with the following code:
 
   [!code-cshtml[Index](get-started-signalr-core/sample/Pages/Index.cshtml)]
 
   The preceding HTML displays name and message fields, and a submit button. Notice the script references at the bottom: a reference to SignalR and *chat.js*.
 
-1. Add a JavaScript file to the *wwwroot\js* folder named *chat.js* and add the following code to it:
+1. Add a JavaScript file to the *wwwroot\js* folder named *chat.js*, and add the following code to it:
```

Comment:
~~Add a JavaScript file to the *wwwroot\js* folder named *chat.js*, and add the following code to it:~~
Add a JavaScript file, named *chat.js*, to the *wwwroot\js* folder. Add the following code to it: